client/sources/common/core.py

```diff
@@ -0,0 +1,203 @@
+import collections
```

Comment:
Module needs a bit more documentation; module string something like

"""Infrastructure for serialization"

and class strings for `Field` and `Serialize`

The rest seems pretty straightforward to me.